src/sage/combinat/integer_vector.py

```diff
@@ -855,8 +855,7 @@ def __init__(self, n):
         self.n = n
         if self.n == 0:
             IntegerVectors.__init__(self, category=EnumeratedSets())
-        else:
```

Comment:
This `else:` *is* necessary, there is no `return` or `raise` before.

Reply:
These things are _bound to happen_ if one goes else-hunting indiscriminately. It would be good to remember the saying "don't fix it if it isn't broken".

I have heard that some misguided coding standards try to fix "no else after return", indiscriminately, but I have not heard SageMath would be among those.

Reply:
I don't care much about any particular rule, however, I favour uniform coding style.  I don't really see why always having the `else:` would be any better or any worse, both have their pro and cons.  In any case, at least at some point, the linter checked for `else` after `return`, so a fair bit of the codebase adheres to this rule.

I remember that, when I was using Turbo Pascal in the 90s, I spent a night over a forgotten semicolon (or was it a comma?).  Who cares?

Reply:
I have given my advice, based on years of coding, and will leave it at that. Word to the wise, and all that.